File: WoofIOS/SwiftPackages/Data/Repositories/Sources/Repositories/Dogs/Mapper/DogProfileDTOToDomainMapper.swift
import DI
import DTO
import Models

@Factory
public final class DogProfileDTOToDomainMapper {
    private let dogMapper: DogDTOToDomainMapper
    private let sizeMapper: DogSizeDTOToDomainMapper

    public func callAsFunction(_ dto: DogProfileDTO) -> DogProfile {
        DogProfile(
            dog: dogMapper(dto.dog),
            breed: dto.breed,
            ageInYears: dto.ageInYears,
            size: sizeMapper(dto.size),
            neighborhood: dto.neighborhood,
            personality: dto.personality,
            favoriteActivity: dto.favoriteActivity,
            bio: dto.bio
        )
    }
}

File: WoofAndroid/data/repositories/src/main/kotlin/com/perrystreet/woof/repositories/dogs/mapper/DogProfileDTOToDomainMapper.kt
package com.perrystreet.woof.repositories.dogs.mapper

import com.perrystreet.woof.dto.dog.DogProfileDTO
import com.perrystreet.woof.models.dog.DogProfile
import org.koin.core.annotation.Factory

@Factory
class DogProfileDTOToDomainMapper(
    private val dogMapper: DogDTOToDomainMapper,
    private val sizeMapper: DogSizeDTOToDomainMapper,
) {
    operator fun invoke(dto: DogProfileDTO): DogProfile = DogProfile(
        dog = dogMapper(dto.dog),
        breed = dto.breed,
        ageInYears = dto.ageInYears,
        size = sizeMapper(dto.size),
        neighborhood = dto.neighborhood,
        personality = dto.personality,
        favoriteActivity = dto.favoriteActivity,
        bio = dto.bio,
    )
}
